packages/kbn-babel-preset/webpack_preset.js

```diff
@@ -6,6 +6,12 @@
  * Side Public License, v 1.
  */
 
+const usesStyledComponents = [
+  /src[\/\\]plugins[\/\\](data|kibana_react)[\/\\]/,
+  /x-pack[\/\\]plugins[\/\\](apm|beats_management|fleet|infra|lists|observability|osquery|security_solution|uptime)[\/\\]/,
+  /x-pack[\/\\]test[\/\\]plugin_functional[\/\\]plugins[\/\\]resolver_test[\/\\]/,
```

Comment:
Is there a useful error message if a developer tries to use styled-components outside these directories?

Reply:
Not currently, but if the ops folks agree that this opt-in/opt-out approach is the right way to go, we can look at adding a message.

Reply:
Is the plan to discouraging use of styled-components and push the teams working on these plugins to migrate to emotion?

What if we added an eslint error message for importing styled-components that told people to use `@emotion/react` instead. Then people would need to disable the eslint rule when importing it which I think is ideal if we're actively discouraging the spread of styled-compononents. We could also have the opposite rule in place and maintain this list of selectors in `.eslintrc.js` or something so they stay synchronized.

(This is quite a list of plugins already using styled-components though, didn't expect it to be so large)

Reply:
> Is the plan to discouraging use of styled-components and push the teams working on these plugins to migrate to emotion?

Yes. The APIs are similar enough that migration shouldn't be time consuming. I was also surprised to see so many plugins using it; the list has grown since I last did an audit.

Good ideas regarding eslint. I'll look in to creating a rule and some kind of synchronizable list.

Reply:
Great, I think since we're going to need to scope this to specific files we might want to customize the `@kbn/eslint/module_migration` rule: https://github.com/elastic/kibana/blob/master/packages/elastic-eslint-config-kibana/.eslintrc.js#L37-L41

We can add file globs to include/exclude specific migration steps for specific patterns using regex like above. Should be as simple as [filtering the list of mappings](https://github.com/elastic/kibana/blob/master/packages/kbn-eslint-plugin-eslint/rules/module_migration.js#L98) and then returning an empty visitor `{}` when the mappings are empty.

Reply:
Should we create a meta issue with an explicit deadline to track the migration process?

Reply:
Finally got back to this. Extended `@kbn/eslint/module_migration` to accept `include` and `exclude` and used that to write a new rule for `styled-components`. The file regex array used by `@kbn/babel-preset` is now stored in `@kbn/dev-utils` and is shared across packages.